src/Form.js

```diff
@@ -2,20 +2,122 @@ import React from "react";
 import "./Form.css";
 
 class Form extends React.Component {
+  constructor(){
+    super()
+    this.state ={
+      input: "",
+      operation: "",
+    }
+  }
+
+  handleInputBox =(event)=> {
+    this.setState ({
+      input: event.target.value
+    })
+  }
+
+  handleOperationChange =(event)=>{
+    this.setState({
+      operation: event.target.value
+    })
+  }
+
+  calcSum =(arr)=>{
+    let sum = 0;
+    for(let num of arr){
+      sum += num;
+    }
+    return sum;
+  }
+
+  calcMode = (arr) => {
+    let countObj = {};
+    for(let num of arr) {
+      //countObj[num] = (countOjb[num] || 0) + 1;
+      if(countObj[num]) {
+        countObj[num] += 1;
+      } else {
+        countObj[num] = 1;
+      }
+    }
+
+    let keys = Object.keys(countObj);
+    let hightValue = 0;
+    let hightestKey;
+    for(let key of keys){
+      if(countObj[key] > hightValue) {
+        hightValue = countObj[key];
+        hightestKey = key;
+      }
+    }
+    return hightestKey;
+  }
+  
+  handleValidation = (numArr) => {
+    let flag = true;
+    if(this.state.input=== ""){
+      flag = false
+    }
+    for(let num of numArr) {
+      if(isNaN(num)) {
+        flag = false;
+      }
+    }
+    return flag;
+  }
+
+  handleSubmit =(event)=>{
+    event.preventDefault();
+    let newArr = this.state.input.split(",");
+    let numArr = newArr.map(str => Number(str));
+    let isValid = this.handleValidation(numArr);
+
+    if(this.state.operation === "") {
+      this.props.handleChangeResult("Invalid operation.");
```

Comment:
The only reason your tests aren't passing is because of this line. When the cypress tests run, they test empty and invalid inputs with a **blank operation** too. Since you check for "Invalid operation" **before** you check for "Invalid input", the error message that ends up being read is "Invalid operation.", which is NOT what the tests want to see. They want to see "Invalid input."